packages/context/src/BindingKey.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+export type BindingAddress<T> = string | BindingKey<T>;
+
+// tslint:disable-next-line:no-unused-variable
+export class BindingKey<ValueType> {
+  static readonly PROPERTY_SEPARATOR = '#';
+
+  /**
+   * Create a new key for a binding bound to a value of type `ValueType`.
+   *
+   * @param key The binding key.
+   */
+  public static create<ValueType>(key: string): BindingKey<ValueType> {
+    return new BindingKey<ValueType>(key);
```

Comment:
Let's change `key` to `keyWithPath` and parse it into `{key, path?}`.

Reply:
I'll allow the following flavors:

```ts
BindingKey.create('rest');
BindingKey.create('rest#port');
BindingKey.create('rest', 'port');
```